### DISCUSSION TITLE
Read configuration without requiring plugin and extension resolution

### DISCUSSION THREAD

**Discussion by knutwannheden** [eslint/eslint#17979] [Q&A] [answered]
I was wondering if there is any way to read the "raw" unresolved configuration using the ESLint API. I couldn't find any options, so I suspect this isn't possible.

My use case is that I would like to run ESLint across a project with the rules being defined externally. I tried working with config overrides, but AFAICT that will still pull in the plugins defined in the project, unless I explicitly opt out from reading the project's RC file. But if I do that then I don't have access to the project's parser and env config. So I would like to verify that a large set of projects all conform to a given set of rules. Is there maybe some way I can achieve that which doesn't require me to always resolve the project's plugins (and thus potentially download a lot of NPM packages)?

**Comment by nzakas** [comment, marked as answer]
No, this isn't possible. If you're using a `.eslintrc` file, then you can just read in the file and parse it yourself. But any config is unusable by ESLint without resolving it first.

**Reply by knutwannheden** [reply]
I suspected as much. AFAICT the `useEslintrc` parameter also applies to all other supported config file formats. Since I don't want to implement parsing for all of these, I will have to think of something else.

Do you think my use case sounds reasonable? Can you think of some other way of achieving this?
